model/listener/CentralSyncLogListener.php

```diff
@@ -0,0 +1,193 @@
+        } catch (Exception $e) {
+            $this->logError($e->getMessage());
+        }
+    }
+
+    /**
+     * Update synchronization response record.
+     *
+     * @param AbstractSyncEvent $event
+     * @throws common_exception_Error
+     */
+    private function updateSyncLogRecord(AbstractSyncEvent $event)
+    {
+        $params = $event->getSyncParameters();
+        $this->validateParameters($params);
+        $syncLogService = $this->getSyncLogService();
+
+        /** @var SyncLogEntity $syncLogEntity */
+        $syncLogEntity = $syncLogService->getBySyncIdAndBoxId($params[SyncLogServiceInterface::PARAM_SYNC_ID], $params[SyncLogServiceInterface::PARAM_BOX_ID]);
+
+        $report = $syncLogEntity->getReport();
+        $report->add($event->getReport());
+
+        $eventData = $this->parseSyncData($event->getReport());
+        $newSyncData = SyncLogDataHelper::mergeSyncData($syncLogEntity->getData(), $eventData);
+        $syncLogEntity->setData($newSyncData);
+
+        $syncLogService->update($syncLogEntity);
+    }
+
+    /**
+     * Create synchronization log record.
+     *
+     * @param SyncRequestEvent $event
+     * @throws common_exception_Error
+     */
+    private function createSyncLogRecord(SyncRequestEvent $event)
+    {
+        $params = $event->getSyncParameters();
+        $this->validateParameters($params);
+        $syncLogService = $this->getSyncLogService();
+
+        $report = Report::createInfo('Synchronization started...');
+        $report->add($event->getReport());
+        $syncLogEntity = new SyncLogEntity(
+            $params[SyncLogServiceInterface::PARAM_SYNC_ID],
+            $params[SyncLogServiceInterface::PARAM_BOX_ID],
+            $params[SyncLogServiceInterface::PARAM_ORGANIZATION_ID],
+            $this->parseSyncData($report),
+            SyncLogEntity::STATUS_IN_PROGRESS,
+            $report,
+            new DateTime()
+        );
+
+        $syncLogService->create($syncLogEntity);
+    }
+
+    /**
+     * @param SyncFinishedEvent $event
+     */
+    public function logSyncFinished(SyncFinishedEvent $event)
+    {
+        try {
+            $params = $event->getSyncParameters();
+            $this->validateParameters($params);
+            $syncLogService = $this->getSyncLogService();
+
+            /** @var SyncLogEntity $syncLogEntity */
+            $syncLogEntity = $syncLogService->getBySyncIdAndBoxId($params[SyncLogServiceInterface::PARAM_SYNC_ID], $params[SyncLogServiceInterface::PARAM_BOX_ID]);
+
+            $report = $syncLogEntity->getReport();
+            $report->add($event->getReport());
+            if ($report->containsError()) {
+                $syncLogEntity->setFailed();
+            } else {
+                $syncLogEntity->setCompleted();
+            }
+            $syncLogEntity->setFinishTime(new DateTime());
+
+            $syncLogService->update($syncLogEntity);
+        } catch (Exception $e) {
+            return;
```

Comment:
why we need this return?

Reply:
It's just personal preference. Do you prefer something like this: `} catch (Exception $e) {}` ?